### PR TITLE
Remove lintcheck log and add `--filter` option to lintcheck

### PR DESCRIPTION
**Pull request by xFrednet**
This PR removes the really outdated lintcheck log from this repo and adds a new `--filter` option to force warnings of a lint. This also ignores `allow` attributes. I believe that this is a good thing, as we'll see if suppressed false positives are also fixed by the changes.

---

#### Example:
```sh
$ cargo lintcheck --filter identity_op
```

#### Output
```
clippy 0.1.59 (460bef22a 2021-12-08)

[..]/cargo-0.49.0/src/cargo/core/compiler/fingerprint.rs:1910:17 clippy::identity_op "the operation is ineffective. Consider reducing it to `(ret[0] as usize)`"
[..]/cargo-0.49.0/src/cargo/util/hex.rs:8:9 clippy::identity_op "the operation is ineffective. Consider reducing it to `num`"
[..]/libc-0.2.81/src/unix/linux_like/linux/mod.rs:2654:18 clippy::identity_op "the operation is ineffective. Consider reducing it to `(dev & 0x00000000000000ff)`"
[..]/libc-0.2.81/src/unix/linux_like/linux/mod.rs:2665:16 clippy::identity_op "the operation is ineffective. Consider reducing it to `(minor & 0x000000ff)`"
[..]/libc-0.2.81/src/unix/linux_like/mod.rs:1218:27 clippy::identity_op "the operation is ineffective. Consider reducing it to `IPOPT_CONTROL`"
[..]/rayon-1.5.0/src/slice/quicksort.rs:588:17 clippy::identity_op "the operation is ineffective. Consider reducing it to `len / 4`"
[..]/regex-1.3.2/src/dfa.rs:1380:14 clippy::identity_op "the operation is ineffective. Consider reducing it to `(empty_flags.start as u8)`"

Stats:
clippy::identity_op 7
ICEs:
```
Two of these lints are suppressed when lintcheck runs normally

---

changelog: none

r? @matthiaskrgr Feel free to reassign if you think that someone else would be a better reviewer :upside_down_face:

### CODE REVIEW

File: lintcheck/src/main.rs [hunk 1]
```diff
@@ -410,12 +420,26 @@ impl LintcheckConfig {
...
+                iter.map(|lint_name| {
+                    let mut filter = lint_name.replace('_', "-");
+                    if !filter.starts_with("clippy::") {
+                        filter.insert_str(0, "clippy::");
+                    }
+                    filter
+                })
+                .collect()
+            })
+            .unwrap_or_else(|| Vec::new());
```
Comment (flip1995) [inline lintcheck/src/main.rs:435]:
```suggestion
            .unwrap_or_default();
```

**Review by flip1995 — COMMENTED** [review]
LGTM overall, just a NIT. Waiting for @matthiaskrgr review though

**Review by matthiaskrgr — APPROVED** [review]
Yeah, I didn't get to have the time or motivation to actually monitor clippy lint changes as intended to back then :upside_down_face: 
It doesn't really make sense to carry around an outdated logfile all the time.

And we actually have clippy lint results fed into the lintcheck repo by now which didn't exist back then  https://github.com/matthiaskrgr/clippy-lintcheck/commits/main

By default we still write everything out into a file and do not filter out anything, right?

**Comment by xFrednet** [comment]
The logs will remain as they are, the filter only applies if the `--filter` option is added.

**Comment by matthiaskrgr** [comment]
Ok great :)
@bors r+

**Pull request merged**